service/controller/remotewrite/resource.go

```diff
@@ -28,8 +29,23 @@ func newResources(config ControllerConfig) ([]resource.Interface, error) {
 		}
 	}
 
+	var rwSecretResource resource.Interface
+	{
+		c := remotewritesecret.Config{
+			K8sClient:        config.K8sClient,
+			Logger:           config.Logger,
+			PrometheusClient: config.PrometheusClient,
+		}
+
+		rwSecretResource, err = remotewritesecret.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
 	resources := []resource.Interface{
 		prometheusRemoteWrite,
+		rwSecretResource,
```

Comment:
I think `rwSecretResource` should go first as its better to create secrets before configuring the remote write field.

```suggestion
		rwSecretResource,
		prometheusRemoteWrite,
```